### ISSUE TITLE
Can't stop dependency logging

### ISSUE THREAD

**Issue by chocvanstraw** [Azure/azure-functions-dotnet-worker#2225]
We recently upgraded a .net 6 inprocess function to a .net 8 isolated function.  The function has a service bus trigger that pulls a message from a topic and sends the data to an api.  We only want to see our custom trace logs in app insights.  We've configured logging so that everything is set to Error except our custom logging which is set to Information. We have also set the EnableDependencyTrackingTelemetryModule = false in Program.cs

However, every time the function executes, I see the following dependency log in app insights (see attached).  I can't tell if it's coming from the function, service bus or http client that makes the api call but its effectively doubling our app insights logs as we can't turn it off no matter what combination of settings we try.  This never appeared until the upgrade to .NET 8 isolated.

<img width="801" alt="AppInsightsLog" src="https://github.com/Azure/azure-functions-dotnet-worker/assets/20091923/77d2bb1f-12ba-4366-b9fb-1cc43c6e72c6">

Has anyone else experienced this?

Thanks for any help.

Here is the host.json:

{
  "version": "2.0",
  "logging": {
    "logLevel": {
      "default": "Error",
      "Microsoft": "Error",
      "Worker": "Error",
      "System": "Error",
      "Host.Aggregator": "Error",
      "Host.Results": "Error",
      "Azure.Core": "Error",
      "Azure.Messaging.ServiceBus": "Error",
      "Function": "Error",
      "Function.XXX": "Error",
      "Function.XXX.User": "Information"
    },
    "applicationInsights": {
      "samplingSettings": {
        "isEnabled": true,
        "excludedTypes": "Exception"
      },
      "enableLiveMetricsFilters": true,
      "enableDependencyTracking": false
    }
  }
}

**Comment by kshyju** [comment]
Are you logging to AI from the isolated function app like this https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide?tabs=windows#application-insights ?

**Comment by chocvanstraw** [comment]
Yes. Our custom logs, which use the .net logger show up fine in AI but we always get this rogue dependency log show up.  We disabled dependencies in the AI section of host.json but that didn’t do anything. We ended up disabling dependencies in program.cs which reduced dependency logs significantly except for the one shown in the screen shot.ThanksOn Jan 19, 2024, at 4:30 PM, Shyju Krishnankutty ***@***.***> wrote:﻿
Are you logging to AI from the isolated function app like this https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide?tabs=windows#application-insights ?

**Comment by kshyju** [comment]
@RohitRanjanMS

**Comment by RohitRanjanMS** [comment]
Hi @chocvanstraw, "Invoke" dependency is emmited by the Functions Worker process. You can add a telemetry processor to skip it.
Here's an example https://github.com/RohitRanjanMS/ApplicationInsightsDemo/blob/main/Isolated/IsolatedApplicationInsightsDemo/CustomTelemetryProcessor.cs
